source/connect/csot.txt

```diff
@@ -0,0 +1,234 @@
+
+      .. code-block:: ruby
+         :emphasize-lines: 2
+
+         uri = "<connection string>"
+         options = { timeout_ms: 30000 }
+
+         client = Mongo::Client.new(uri, options)
+   
+   .. tab:: Connection String
+      :tabid: connection-string
+
+      .. code-block:: ruby
+         :emphasize-lines: 1
+
+         uri = "mongodb://<hostname>:<port>?timeoutMS=30000"
+         client = Mongo::Client.new(uri)
+
+If you specify the ``timeout_ms`` option, the driver automatically applies the
+specified timeout per each server operation.
+
+.. note:: 
+
+   The ``timeout_ms`` connection option unifies most timeout related options.
+   As of the {+driver-short+} v2.21, the following timeout options are deprecated: 
+
+   - ``socket_timeout``
+   - ``wait_queue_timeout``
+   - ``wtimeout``
+   - ``max_time_ms``
+   - ``max_commit_time_ms``
+
+   The ``timeout_ms`` connection option takes precedence over these 
+   deprecated timeout options. 
+
+Timeout Inheritance
+~~~~~~~~~~~~~~~~~~~
+
+When you specify a ``timeout_ms`` option, the driver applies the timeout
+according to the same inheritance behaviors as the other {+driver-short+} options.
+The following table describes how the timeout value is inherited at each level:
+
+.. list-table::
+   :header-rows: 1
+   :widths: 30 70
+
+   * - Level
+     - Inheritance Description
+
+   * - Operation
+     - Takes the highest precedence and will override ``timeout_ms``
+       options set at any other level.
+
+   * - Transaction
+     - Takes precedence over ``timeout_ms`` set at the session,
+       collection, database, or client level.
+
+   * - Session
+     - Applies to all transactions and operations within
+       that session, unless the option is overridden by options set at those levels.
+
+   * - Database
+     - Applies to all sessions and operations within that
+       database, unless the option is overridden by options set at those levels.
+
+   * - Collection
+     - Applies to all sessions and operations on that
+       collection, unless the option is overridden by options set at those levels.
+
+   * - Client
+     - Applies to all databases, collections, sessions, transactions, and
+       operations within that client that do not otherwise specify
+       ``timeout_ms``.
+
+For more information on overrides and specific options, see the :ref:`Overrides
+<ruby-csot-overrides>` section.
+
+.. _ruby-csot-overrides:
+
+Overrides
+---------
+
+You can specify a ``timeout_ms`` option at the operation level to override the
+client-level configuration for a specific operation. This allows you to
+customize timeouts based on the needs of individual queries.
+
+The following example demonstrates how an operation-level ``timeout_ms``
+configuration can override a client-level ``timeout_ms`` configuration:
+
+.. literalinclude:: /includes/connect/csot.rb
+   :language: ruby
+   :start-after: start-csot-overrides
+   :end-before: end-csot-overrides
+
+Transactions
+~~~~~~~~~~~~
+
```

Comment:
S: add an introductory sentence here like "You can also set timeout options for transactions."